src/index.ts

```diff
@@ -583,4 +585,35 @@ export class ReadonlyArrayType<RT extends Any> implements Type<ReadonlyArray<Typ
 export const readonlyArray = <RT extends Any>(type: RT, name?: string): ReadonlyArrayType<RT> =>
   new ReadonlyArrayType(type, name)
 
+export class StrictType<RT extends InterfaceType<any>> implements Type<TypeOf<RT>> {
+  readonly _tag: 'StrictType' = 'StrictType'
+  readonly _A: TypeOf<RT>
+  readonly validate: Validate<TypeOf<RT>>
+  constructor(readonly type: RT, readonly name: string = `StrictType<${type.name}>`) {
+    const len = Object.keys(type.props).length
+    this.validate = (v, c) =>
+      type.validate(v, c).chain(o => {
+        const keys = Object.keys(v)
```

Comment:
I'm not sure if Object.keys is the correct function here? I'm not certain how symbols and non-enumerable properties should be handled. This is enough for me as all my data is deserialized from JSON before, so it can't contain any symbols or non-enumerable properties, but it might not be enough for the generic case.

Reply:
Also, shouldn't it be `Object.keys(o)` instead of `Object.keys(v)`?

Reply:
> This is enough for me

For me too

> shouldn't it be Object.keys(o)

Right, thanks